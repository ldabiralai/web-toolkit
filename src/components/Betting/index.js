import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import { rgba } from 'polished';
import { coreNeutral9, coreLightMinus1, martinique, manatee } from '../../colors';
import { fontFamilies } from '../../typography';
import { breakpoints } from '../..';

const StyledCard = styled.div`
  background-color: ${coreNeutral9};
  color: ${coreLightMinus1};
  display: flex;
  flex-direction: column;
  flex-grow: 0;
  justify-content: center;
  align-items: center;
  border-radius: 4px;
  position: relative;
  overflow: hidden;
  &::before {
    content: '';
    position: absolute;
    top: -35px;
    left: 0;
    right: 0;
    display: block;
    height: 60px;
    background-color: ${martinique};
    transform: skew(0, -5deg);
    z-index: 1;
  }
`;

const StyledBettingName = styled.div`
  background-color: ${props => props.background};
  color: ${props => props.text};
  padding: 13px;
  text-transform: uppercase;
  font-weight: 500;
  letter-spacing: 1px;
  z-index: 2;
`;

const StyledBettingSentence = styled.div`
  font-family: ${fontFamilies.alphaHeadline};
  text-transform: uppercase;
  font-weight: 500;
  font-size: 18px;
  margin-top: 22px;
  margin-bottom: 4px;
`;

const StyledBetting = styled.div`
  display: flex;
  align-items: flex-start;
  font-family: ${fontFamilies.alphaHeadline};
  line-height: 30px;
  margin-bottom: 35px;
  justify-content: left;
`;

const StyledChoiceWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
`;

const StyledChoiceNumber = styled.div`
  font-size: 13px;
  color: ${manatee};
`;
const StyledChoiceCote = styled.a`
  display: block;
  text-decoration: inherit;
  color: inherit;
  padding: 6px 20px 4px 20px;
  font-size: 24px;
  background-color: ${rgba(coreLightMinus1, 0.2)};
  margin: 0 8px;
  border-radius: 3px;
  &:hover {
    background-color: ${rgba(coreLightMinus1, 0.4)};
  }
  ${breakpoints.medium(css`
    margin: 0 3px;
  `)}
`;

const StyledTeamName = styled.div`
  text-transform: uppercase;
  font-weight: normal;
  ${breakpoints.medium(css`
    line-height: 42px;
    position: absolute;
    top: 30px;
  `)};
  ${StyledChoiceWrapper}:first-child & {
    ${breakpoints.medium(css`
      right: 100%;
      margin-right: 12px;
    `)}
  }

  ${StyledChoiceWrapper}:last-child & {
    ${breakpoints.medium(css`
      left: 100%;
      margin-left: 12px;
    `)}
  }
`;

export const getRandomSentence = sentences => sentences[Math.floor(Math.random() * sentences.length)];

const Betting = ({ sponsor, sentences, choices }) => (
  <StyledCard>
    <StyledBettingName background={sponsor.backgroundColor} text={sponsor.textColor}>
      {sponsor.name}
    </StyledBettingName>
    <StyledBettingSentence>{getRandomSentence(sentences)}</StyledBettingSentence>
    <StyledBetting>
      {choices.map(choice => (
        <StyledChoiceWrapper key={choice.number}>
          <StyledChoiceNumber>{choice.number}</StyledChoiceNumber>
          <StyledChoiceCote href={choice.link} target="_blank">
            {choice.cote}
          </StyledChoiceCote>
          {choice.label && <StyledTeamName>{choice.label}</StyledTeamName>}
        </StyledChoiceWrapper>
      ))}
    </StyledBetting>
  </StyledCard>
);

Betting.propTypes = {
  sponsor: PropTypes.shape({
    name: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    textColor: PropTypes.string.isRequired,
  }).isRequired,
  sentences: PropTypes.arrayOf(PropTypes.string).isRequired,
  choices: PropTypes.arrayOf(
    PropTypes.shape({
      number: PropTypes.number.isRequired,
      cote: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired,
      label: PropTypes.string,
    })
  ).isRequired,
};

export default Betting;
