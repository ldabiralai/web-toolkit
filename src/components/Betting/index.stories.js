import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import styled from 'react-emotion';
import { Betting } from '../..';

const betting = storiesOf('Components|Betting', module).addDecorator(withInfo);

const Wrapper = styled.div`
  max-width: 500px;
`;

const doubleChoiceData = {
  sponsor: {
    name: 'Betline',
    backgroundColor: '#b4292c',
    textColor: '#ffffff',
  },
  sentences: ['Pariez !', 'PARIEZ AVEC Betline'],
  choices: [
    { number: 1, cote: 2.8, label: 'Nadal', position: 'left' },
    { number: 2, cote: 1.7, label: 'Federer', position: 'right' },
  ],
};

const tripleChoiceData = {
  ...doubleChoiceData,
  choices: [
    { number: 1, cote: 2.8, label: 'Nadal', position: 'left' },
    { number: 2, cote: 4.9 },
    { number: 3, cote: 1.7, label: 'Federer', position: 'right' },
  ],
};

const alternativeLogo = {
  ...doubleChoiceData,
  sponsor: {
    name: 'AlterBet',
    backgroundColor: '#6decb9',
    textColor: '#3c3c3c',
  },
  sentences: ['Alternate with AlterBet !', 'AlterBet for the BETter'],
};

betting
  .add('Double choice', () => (
    <Wrapper>
      <Betting {...doubleChoiceData} />
    </Wrapper>
  ))
  .add('Triple choice', () => (
    <Wrapper>
      <Betting {...tripleChoiceData} />
    </Wrapper>
  ))
  .add('Alternative brand', () => (
    <Wrapper>
      <Betting {...alternativeLogo} />
    </Wrapper>
  ));
