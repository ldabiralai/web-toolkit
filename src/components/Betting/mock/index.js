export const doubleChoiceData = {
  sponsor: {
    name: 'Betline',
    backgroundColor: '#b4292c',
    textColor: '#ffffff',
  },
  sentences: ['Pariez !', 'PARIEZ AVEC Betline'],
  choices: [
    { number: 1, cote: '2.80', label: 'Nadal', position: 'left', link: 'http://google.fr' },
    { number: 2, cote: '1.70', label: 'Federer', position: 'right', link: 'http://bing.fr' },
  ],
};

export const tripleChoiceData = {
  ...doubleChoiceData,
  choices: [
    { number: 1, cote: '2.80', label: 'Nadal', position: 'left', link: 'http://google.fr' },
    { number: 2, cote: '4.90', link: 'http://yahoo.fr' },
    { number: 3, cote: '1.70', label: 'Federer', position: 'right', link: 'http://bing.fr' },
  ],
};

export const alternativeLogo = {
  ...doubleChoiceData,
  sponsor: {
    name: 'AlterBet',
    backgroundColor: '#6decb9',
    textColor: '#3c3c3c',
  },
  sentences: ['Alternate with AlterBet !', 'AlterBet for the BETter'],
};
