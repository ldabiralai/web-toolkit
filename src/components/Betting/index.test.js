import { shallow } from 'enzyme/build';
import React from 'react';
import Betting, { getRandomSentence } from '.';
import { doubleChoiceData, tripleChoiceData } from './mock';

describe('Betting', () => {
  beforeEach(() => {
    const mockMath = Object.create(global.Math);
    mockMath.random = () => 0;
    global.Math = mockMath;
  });
  it('renders Betting for double choices', () => {
    expect(shallow(<Betting {...doubleChoiceData} />)).toMatchSnapshot();
  });

  it('renders Betting for triple choices', () => {
    expect(shallow(<Betting {...tripleChoiceData} />)).toMatchSnapshot();
  });
  it('should return a random sentence', () => {
    expect(getRandomSentence(['a', 'b'])).toEqual('a');
    global.Math.random = () => 0.8;
    expect(getRandomSentence(['a', 'b'])).toEqual('b');
  });
});
