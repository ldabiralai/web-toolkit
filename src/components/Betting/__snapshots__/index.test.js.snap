// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Betting renders Betting for double choices 1`] = `
<Styled(div)>
  <Styled(div)
    background="#b4292c"
    text="#ffffff"
  >
    Betline
  </Styled(div)>
  <Styled(div)>
    Pariez !
  </Styled(div)>
  <Styled(div)>
    <Styled(div)
      key="1"
    >
      <Styled(div)>
        1
      </Styled(div)>
      <Styled(a)
        href="http://google.fr"
        target="_blank"
      >
        2.80
      </Styled(a)>
      <Styled(div)>
        Nadal
      </Styled(div)>
    </Styled(div)>
    <Styled(div)
      key="2"
    >
      <Styled(div)>
        2
      </Styled(div)>
      <Styled(a)
        href="http://bing.fr"
        target="_blank"
      >
        1.70
      </Styled(a)>
      <Styled(div)>
        Federer
      </Styled(div)>
    </Styled(div)>
  </Styled(div)>
</Styled(div)>
`;

exports[`Betting renders Betting for triple choices 1`] = `
<Styled(div)>
  <Styled(div)
    background="#b4292c"
    text="#ffffff"
  >
    Betline
  </Styled(div)>
  <Styled(div)>
    Pariez !
  </Styled(div)>
  <Styled(div)>
    <Styled(div)
      key="1"
    >
      <Styled(div)>
        1
      </Styled(div)>
      <Styled(a)
        href="http://google.fr"
        target="_blank"
      >
        2.80
      </Styled(a)>
      <Styled(div)>
        Nadal
      </Styled(div)>
    </Styled(div)>
    <Styled(div)
      key="2"
    >
      <Styled(div)>
        2
      </Styled(div)>
      <Styled(a)
        href="http://yahoo.fr"
        target="_blank"
      >
        4.90
      </Styled(a)>
    </Styled(div)>
    <Styled(div)
      key="3"
    >
      <Styled(div)>
        3
      </Styled(div)>
      <Styled(a)
        href="http://bing.fr"
        target="_blank"
      >
        1.70
      </Styled(a)>
      <Styled(div)>
        Federer
      </Styled(div)>
    </Styled(div)>
  </Styled(div)>
</Styled(div)>
`;
